Cleanup MC/DC intrinsics for #82448

chapuni opened the pull request:
3rd arg of `tvbitmap.update` was made unused. Remove 3rd arg.

Sweep `condbitmap.update`, since it is no longer used.

chapuni committed:
Cleanup MC/DC intrinsics for #82448

3rd arg of `tvbitmap.update` was made unused. Remove 3rd arg.

Sweep `condbitmap.update`, since it is no longer used.

evodius96 reviewed (approved):
LGTM thanks

zmodem commented:
> Re. condbitmap.update, it has become useless after #82448 . I think pruning it will be the possible way.

Can you expand on this? I'm not familiar with these intrinsics at all. What are users who were using `condbitmap.update` supposed to call now?

chapuni commented:
@zmodem #82448 doesn't use intermediate condbitmap but integer index. I decided not to modify and use the intrinsic `condbitmapupdate`, since the signature will be quite different and this is too simple to be put into the intrinsic.
Also pre-#82448 impl of `condbitmapupdate` was so simple (`|= (V << ID)`) just to update the local variable. I wondered why it was implemented with the intrinsic.

In contrast, I modified `tvbitmapupdate` in #82448 without getting rid of it. I modified its signature and semantics. I thought `tvbitmapupdate` might be left as an intrinsic since it could be enhanced to atomic ops (and continuous bias mode).

I guess they would hit apparently buggy behavior if they were using obsolete `condbitmapupdate` and modified `tvbitmapupdate`.

Since I didn't update and won't update the semantics of `condupdate`, they should get rid of it, as far as they follow our trunk. I don't think I could provide "easy" migration path.

chapuni commented:
I've created #95887 (Release notes).
